### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Markolie**
Fixes the DNA vault station goal not completing properly.
Fixes (advanced) ghost interaction with mineral doors, newscasters, fire alarms and ATM's.
Fixes some stray fruit at the Syndicate base on Z2.
Fixes an incorrect camera network in the courtroom.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6621.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6684.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6642.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6646.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6656.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6657.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6671.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6685.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6677.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6702.
Fixes https://github.com/ParadiseSS13/Paradise/issues/6712.

Fixes https://github.com/ParadiseSS13/Paradise/issues/6725
Matter grippers can now carry firelock electronics as well as bluespace crystals (for construction).

Fixes https://github.com/ParadiseSS13/Paradise/issues/6593.
Cortical speech now uses "bo".

Fixes https://github.com/ParadiseSS13/Paradise/issues/6625
Our atmos pipes don't really support being put on the map as a fitting, so I've just removed it as fixing this issue would not be worth the effort.

Fixes https://github.com/ParadiseSS13/Paradise/issues/6645
Attempting to blow up an undetonatable PDA now shows a warning message. It will also no longer use up a charge. This also fixes an issue where the hidden_uplink PDA check did not work properly.

Also refactors the shitty mineral door code. Animals can now open them and they can no longer be clsoed when there's a mob inside.

🆑 Markolie
fix: The DNA Vault station goal will now be marked as completed properly.
fix: Multi-tile airlocks will now change their direction properly.
fix: The atmos skirt loadout item will now be equipped properly on round start.
fix: Fixes the inhand disabler sprite pointing the wrong way in some directions.
fix: The Janicart will no longer act as a jetpack.
fix: You can now resist out of a borer's control again.
fix: Fixes the general antag ban not working on certain conversion antagonists.
fix: Fixes removing the trunk from underneath a chute/bin causing all sorts of issues.
fix: Fixes alt titles not showing up on ID's.
fix: Mineral doors no longer block atmos (like they're supposed to).
fix: Fixes the cortical borer hivemind talk not working.
fix: Fixes detomatix cartridges not working *from* PDA's with an uplink (it now no longer *on* a PDA with an uplink to prevent blowing up fellow traitors).
fix: Fixes NODROP interaction with secure closets.
fix: Fixes some stray fruit at the Syndicate base on Z2.
fix: Fixes an incorrect camera network in the courtroom.
fix: Fixes an issue where AI's would go blind upon travelling into space.
tweak: The cortical borer hivemind talk key is now "bo".
tweak: Attempting to blow up an undetonatable PDA (such as the captain's) will now show a warning message. It will also no longer use up a charge.
tweak: Animals can now open mineral doors and mineral doors will no longer be closable if there's a mob inside.
tweak: Matter grippers can now carry firelock electronics as well as bluespace crystals (for construction).
rscdel: Removes the pipe fitting on the table from atmospherics as it was broken.
/🆑

### CODE REVIEW

**Comment by Anasari** [comment]
You're a saviour

File: code/game/objects/structures/door_assembly.dm [hunk 1]
```diff
@@ -115,12 +115,6 @@
 	airlock_type = "/multi_tile/glass"
 	glass = -1 //To prevent bugs in deconstruction process.
 
-/obj/structure/door_assembly/door_assembly_cult
```
Comment (Markolie) [inline code/game/objects/structures/door_assembly.dm]: This was a duplicate definition.

**Comment by DesolateG** [comment]
@Markolie 

"Our atmos pipes don't really support being put on the map as a fitting" 

i'm confused.

**Comment by Markolie** [comment]
Our atmos/piping system does not support the pipe item (the one you get from the dispenser) being put on the map by editing the map file - for it to work it needs to be spawned in in-game.

**Comment by KasparoVy** [comment]
What a beast.

**Comment by taukausanake** [comment]
> The only good bug's a dead bug!

Inb4 that is taken out of context for Kidan

File: code/game/machinery/firealarm.dm [hunk 2]
```diff
@@ -232,35 +232,35 @@ FIRE ALARM
...
 	update_icon()
-	//playsound(src.loc, 'sound/ambience/signal.ogg', 75, 0)
+	//playsound(loc, 'sound/ambience/signal.ogg', 75, 0)
 	return
 
 /obj/machinery/firealarm/New(loc, dir, building)
 	..()
 
 	if(loc)
-		src.loc = loc
```
Comment (Fox-McCloud) [inline code/game/machinery/firealarm.dm]: This is going to break things; this is a legitimate instance of implied `src` being necessary. 

Careful when removing `src` to check for things like this.

File: code/game/gamemodes/miniantags/borer/borer.dm [hunk 3]
```diff
@@ -145,7 +145,7 @@
 	if(client.statpanel == "Status")
 		stat("Chemicals", chemicals)
 
-/mob/living/simple_animal/borer/say(message)
+/mob/living/simple_animal/borer/say(var/message)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/miniantags/borer/borer.dm]: no need for implied `var` in arguments.

**Comment by Fox-McCloud** [comment]
Just a couple of things.

@Crazylemon64

**Comment by Markolie** [comment]
Two more fixes: Fixed a NODROP issue with secure lockers and stray fruit at the Syndicate base on Z2.

**Comment by Fox-McCloud** [comment]
@Markolie 

The stray fruit looks like it came from the admin room; is it still there?

**Comment by Markolie** [comment]
It is.

**Comment by Jovaniph** [comment]
Damnit you did it already. I was going to do exactly that. Just editing the file itself...fuuuuck. q.q

**Comment by SamHPurp** [comment]
@Markolie @Fox-McCloud The fruit table appeared to appear at around the time the PR that allowed hijacked shuttles to dock at the Syndie base.

File: code/game/machinery/firealarm.dm [hunk 4]
```diff
@@ -232,35 +232,35 @@ FIRE ALARM
...
 	..()
 
-	if(loc)
-		src.loc = loc
+	if(location)
+		loc = location
 
 	if(dir)
-		src.dir = dir
+		dir = dir
```
Comment (Fox-McCloud) [inline code/game/machinery/firealarm.dm]: Won't this cause problems, as well?

Comment (Markolie) [inline code/game/machinery/firealarm.dm]: Whoops.

**Comment by Fox-McCloud** [comment]
conflicts; will merge this once those are resolved.

**Pull request merged**